lib/cc/engine/analyzers/file_thread_pool.rb

```diff
@@ -14,11 +14,13 @@ def initialize(files, concurrency: DEFAULT_CONCURRENCY)
 
         def run(&block)
           queue = build_queue
+          lock = Mutex.new
 
           @workers = thread_count.times.map do
             Thread.new do
-              while !queue.empty? && (item = queue.pop(true))
-                yield item
+              until queue.empty?
+                item = lock.synchronize { queue.pop(true) unless queue.empty? }
+                yield item if item
```

Comment:
> WDYT of a small utility method:

That sounds good to me, but I think the lock would need to be a member var, yeah? Getting a new lock on every method call doesn't seem real helpful.

> Another option would be rescue the non-block-empty-queue-popped exception (assuming it's sufficiently specific)

It's a `ThreadError`, and I felt that was too broad to rescue: previously we were rescuing them in the entire thread block. I could rescue *just* the pop, but I'm not confident enough about JRuby threading internals to be sure nothing else could raise that exception class at any given time.